composer/utils/object_store/oci_object_store.py

```diff
@@ -132,17 +142,31 @@ def download_object(
             os.makedirs(dirname, exist_ok=True)
         tmp_path = str(filename) + f'.{uuid.uuid4()}.tmp'
 
+        # Get the size of the object
+        head_object_response = self.client.head_object(self.namespace, self.bucket, object_name)
+        object_size = head_object_response.headers['content-length']
+        # Calculate the part size
+        part_size = -(-int(object_size) // num_parts)  # Ceiling division
+
         try:
-            response = self.client.get_object(
-                namespace_name=self.namespace,
-                bucket_name=self.bucket,
-                object_name=object_name,
-            )
+            # Download parts in parallel
+            args_list = []
+            for i in range(num_parts):
+                start_byte = i * part_size
+                end_byte = min(start_byte + part_size - 1, int(object_size) - 1)
+                args_list.append((object_name, filename, start_byte, end_byte, i))
+            with concurrent.futures.ThreadPoolExecutor() as executor:
+                parts = list(executor.map(lambda p: self._download_part(*p), args_list))
```

Comment:
You should use concurrent.futures.as_completed(.as_completed to write the file parts so they don't all sit in memory unnecessarily and are written as the finish if possible? Unless we have a good reason for this?

Reply:
The reason was to preserve order of the parts for merging. But I can find another solution that sorts them out after.

Reply:
Should be fixed to use `concurrent.futures.as_completed` now.